src/types/types.ts

```diff
@@ -1922,6 +1922,37 @@ export const SortOrder = {
  */
 export type SortOrder = 'asc' | 'desc';
 
+export interface SyntheticDateBase {
+  timestamp?: Date;
+}
+
+export interface SyntheticDataError extends SyntheticDateBase {
+  error: string;
+}
+
+export type SyntheticDataPoint = SyntheticDataValue | SyntheticDataError;
```

Comment:
```suggestion
export type SyntheticDatapoint = SyntheticDataValue | SyntheticDataError;
```
just to be consistent with other places